File: 9.贪心算法/4.最大子序和/53.最大子数组和1.py
#
# @lc app=leetcode.cn id=53 lang=python3
#
# [53] 最大子数组和
#
from mytools import *
# @lc code=start
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        cnt = 0
        res = float('-inf')
        for x in nums:
            cnt += x
            if cnt > res:
                res = cnt
            if cnt < 0: cnt = 0
        return res
# @lc code=end

File: 9.贪心算法/4.最大子序和/53.最大子数组和1.cpp
/*
 * @lc app=leetcode.cn id=53 lang=cpp
 *
 * [53] 最大子数组和
 */
#include "tools.h"
// @lc code=start
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int n = nums.size();
        int res = INT_MIN, tmp = 0;
        for (int i = 0; i < n; i ++ )
        {
            tmp += nums[i];
            res = max(tmp, res);
            if (tmp <= 0) tmp = 0;
        }
        return res;
    }
};
// @lc code=end

